src/Lean/Widget/InteractiveDiagnostics.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Authors: Wojciech Nawrocki
+-/
+import Lean.Data.Lsp
+import Lean.Message
+import Lean.Server.Rpc.Basic
+import Lean.Widget.TaggedText
+import Lean.Widget.ExprWithCtx
+
+namespace Lean.Widget
+open Lsp Server
+
+structure MsgEmbed where
+  -- TODO(WN): other variants, e.g. `ofGoal`
+  ef : ExprText
+  deriving Inhabited, RpcEncoding
```

Comment:
We might want to pick a more descriptive name than `ef`, since this is part of the public RPC API.